vendor/github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1/subscription_types.go

```diff
@@ -109,3 +109,13 @@ func NewInstallPlanReference(ref *corev1.ObjectReference) *InstallPlanReference
 		UID:        ref.UID,
 	}
 }
+
+// SetCatalogSource sets the Spec.source field
+func (s *Subscription) SetCatalogSource(csName string) {
```

Comment:
You cannot modify vendor code like this. This needs a PR against OLM or we carry the code.